docs/HowToFAQ/FAQTypeInference.md

```diff
@@ -0,0 +1,12 @@
+---
+title: I can define a trait with some type parameters say trait Test<A, B, C>. When I use this trait is there a way to get Dafny to infer these types for me?
+---
+
+## Question
+
+I can define a trait with some type parameters say trait `Test<A, B, C>`. When I use this trait is there a way to get Dafny to infer these types for me?
+
+## Answer
+
+Type inference, though quite extensive in its effect, only works within the bodies of functions and methods.
```

Comment:
Sorry about my previous confusion. Here's what I should have showed:

``` dafny
class C<X(0)> {
  static method M() returns (x: X) {
  }

  method P() {
    var x0 := M();        // type of x0 inferred to be X
    var x1 := C<int>.M(); // type of x1 inferred to be int
    var x2: int := C.M(); // M() instantiated with int

    var x3 := C.M();      // error: type of x3 is underspecified
    var x4: int := M();   // error: M() instantiated with X, but x1 has type int
  }
}
```

However, this may be too much detail for the FAQ.

Reply:
Example is helpful. I added text and the example.